### ISSUE TITLE
There is no alt or title text in the PhotoSwipe enlarged product image <img> tag

### ISSUE THREAD

**Issue by cindyward1** [woocommerce/woocommerce#15791]
The particular image tag is `<img class="pswp__img">`. The lack of alt and title text on enlarged images is creating a large number of issues in Raven SEO Tools when it crawls our WooCommerce-based e-commerce site.

The WooCommerce demo using the Storefront theme does not have alt or title text in the product image, so as one would expect, there is no alt or title text in the enlarged image either. Since this is not a theme issue, I tested this issue using our custom theme.

Per [W3Schools](https://www.w3schools.com/tags/att_img_alt.asp), alt text is required: "The required alt attribute specifies an alternate text for an image, if the image cannot be displayed. The alt attribute provides alternative information for an image if a user for some reason cannot view it (because of slow connection, an error in the src attribute, or if the user uses a screen reader)." This is an ADA issue.

The case for a title tag is not as strong, but to quote [Yoast's "alt and title attributes in WordPress" section](https://yoast.com/image-seo-alt-tag-and-title-tag-optimization/), "There’s no excuse for not doing this right, other than laziness. Your (image) SEO will truly benefit if you get these tiny details right and visually challenged users will also like you all the more for it."

## Prerequisites

- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate
- [x] The issue still exists against the latest `master` branch of WooCommerce (_I tested against the latest released version 3.0.9. I do not have a way to test against master. The PhotoSwipe JS files involved have not changed in 5 months._)

**Inspection of enlarged PhotoSwipe product image**
![enlarged photoswipe image](https://user-images.githubusercontent.com/8136992/27556503-b0c814ba-5a6a-11e7-8125-2ca5ec90bfb1.png)

**Inspection of product image**
![product image](https://user-images.githubusercontent.com/8136992/27556502-b0ab22ec-5a6a-11e7-8fda-88cb40dc728b.png)

- [x] This is not a usage question (Those should be directed to the [community](https://wordpress.org/support/plugin/woocommerce), unless this is a question about a premium plugin in which you should [use the helpdesk](https://woocommerce.com/my-account/tickets/) for official extensions or contact the author of 3rd party extensions)
- [x] I have attempted to find the simplest possible steps to reproduce the issue
- [ ] I have included a failing test as a pull request (Optional)

## Steps to reproduce the issue

1. Click on a product image to enlarge it
2. Inspect the enlarged image

## Expected behavior and actual behavior

When I follow those steps, I see...
... no alt or title text in the image <img class="pswp__img">

I was expecting...
... to see alt and title text in the image <img class="pswp_img">

The enlarged image comes from the Media Library just like the product image. The product image has alt and title text but the enlarged image does not.

## Environment

<details>
```
`
### WordPress Environment ###

Home URL: https://www.trialguides.com
Site URL: https://www.trialguides.com/blue32
WC Version: 3.0.8
Log Directory Writable: ✔
WP Version: 4.7.5
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: –
Language: en_US

### Server Environment ###

Server Info: Apache
PHP Version: 7.1.1
PHP Post Max Size: 550 MB
PHP Time Limit: 300
PHP Max Input Vars: 1000
cURL Version: 7.38.0
OpenSSL/1.0.1t

SUHOSIN Installed: –
MySQL Version: 5.7.18
Max Upload Size: 550 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.0.8
WC Database Prefix: wp_trialguides1
woocommerce_sessions: ✔
woocommerce_api_keys: ✔
woocommerce_attribute_taxonomies: ✔
woocommerce_downloadable_product_permissions: ✔
woocommerce_order_items: ✔
woocommerce_order_itemmeta: ✔
woocommerce_tax_rates: ✔
woocommerce_tax_rate_locations: ✔
woocommerce_shipping_zones: ✔
woocommerce_shipping_zone_locations: ✔
woocommerce_shipping_zone_methods: ✔
woocommerce_payment_tokens: ✔
woocommerce_payment_tokenmeta: ✔
MaxMind GeoIP Database: ✔

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (33) ###

Akismet Anti-Spam: by Automattic – 3.3.2
WP Offload S3 - Assets Addon: by Delicious Brains – 1.2.4
WP Offload S3 - Enable Media Replace Addon: by Delicious Brains – 1.0.4
WP Offload S3 - Meta Slider Addon: by Delicious Brains – 1.0.2
WP Offload S3: by Delicious Brains – 1.4
WP Offload S3 - WooCommerce Addon: by Delicious Brains – 1.0.7
Amazon Web Services: by Delicious Brains – 1.0.2
Categories Images: by Muhammad Said El Zahlan – 2.5.3
Clean Notifications: by Mike Davidson – 1.1
Easy FancyBox: by RavanH – 1.6
Enhanced Media Library: by wpUXsolutions – 2.4.5
FD Feedburner Plugin: by John Watson – 1.48
Meta Slider Lightbox: by Matcha Labs – 1.9
Meta Slider: by Matcha Labs – 3.5.1
Ninja Forms - Datepicker: by  –
Ninja Forms - Layout & Styles: by The WP Ninjas – 3.0.19
Ninja Forms: by The WP Ninjas – 3.1.4
Posts 2 Posts: by scribu – 1.6.5
Redirection: by John Godley – 2.6.1
Relevanssi Premium: by Mikko Saari – 1.15.2.1
Analytics for WordPress — by Segment.io: by Segment.io – 1.0.13
ShortPixel Image Optimizer: by ShortPixel – 4.4.2
Tabs In Editor: by Aaron Harun – 1.1
Trial Guides Authors: by Michael Lippold – 1.0.1
WooCommerce Product Bundles: by SomewhereWarm – 5.3.1
WOOF - WooCommerce Products Filter: by realmag777 – 1.1.7
WooCommerce Tab Manager: by SkyVerge – 1.8.1
WooCommerce: by Automattic – 3.0.8 – 3.0.9 is available
WooCommerce Helper: by WooCommerce – 1.7.2 – Network enabled
Yoast SEO: by Team Yoast – 4.9
WP Amazon SQS: by Michael Lippold – 0.0.1
WP Migrate DB Pro Media Files: by Delicious Brains – 1.4.7
WP Migrate DB Pro: by Delicious Brains – 1.7.1

### Settings ###

API Enabled: ✔
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: bundle (bundle)
external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)


### WC Pages ###

Shop base: #11403 - /shop/
Cart: ❌ Page not set
Checkout: ❌ Page not set
My account: ❌ Page not set

### Theme ###

Name: Trial Guides v2
Version: 3.0
Author URL: http://trialguides.com/
Child Theme: ❌ – If you're modifying WooCommerce on a parent theme you didn't build personally
then we recommend using a child theme. See: How to create a child theme

WooCommerce Support: ✔

### Templates ###

Overrides: trialguides-02/woocommerce/archive-product.php
trialguides-02/woocommerce/global/sidebar.php
trialguides-02/woocommerce/global/wrapper-end.php
trialguides-02/woocommerce/global/wrapper-start.php
trialguides-02/woocommerce/loop/add-to-cart.php
trialguides-02/woocommerce/loop/loop-start.php
trialguides-02/woocommerce/loop/orderby.php
trialguides-02/woocommerce/loop/price.php
trialguides-02/woocommerce/single-product/add-to-cart/variable.php
trialguides-02/woocommerce/single-product/add-to-cart/variation-add-to-cart-button.php
trialguides-02/woocommerce/single-product/add-to-cart/variation.php
trialguides-02/woocommerce/single-product/product-attributes.php
trialguides-02/woocommerce/single-product/title.php

`
```
</details>

## Isolating the problem

- [x] This bug happens with only WooCommerce plugin active
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/)
- [x] I can reproduce this bug consistently

**Comment by mikejolley** [comment]
I found a few resources to confirm this before replying, but it's not important. The ACTUAL images output to the page do have alt text and title. It's only the photoswipe image (which appears on click with javascript) which lacks these tags. This image is not important or parsed by search engines.

http://community.theturninggate.net/viewtopic.php?id=7393

> Such things are entirely unimportant to Photoswipe. The presentation is built on-the-fly from Javascript, and doesn't exist until the moment an image is clicked, so it's entirely irrelevant to SEO and sitemaps.

There was also a photoswipe issue confirming the same thing:

https://github.com/dimsemenov/PhotoSwipe/issues/757

CC'ing @claudiosanches and @claudiulodro to commit this to memory in case it comes up again :)

Thanks

**Comment by cindyward1** [comment]
There *IS* a problem, but it isn't what I thought it was when I previously sent in the issue. The good news is that it can be fixed in in the Media Library.

The problem's root lies in woocommerce/templates/single-product/product-image.php. At line 27, $image_title is set to the post_excerpt for the post with ID $post_thumbnail_id. I discovered that in the Media Library, post_excerpt maps to the caption field, NOT the title field. Since we weren't populating the caption field, the title (data-o_title clause of the img tag) ended up blank, which is why Raven SEO Tools gave me a huge number of image issues after we updated to WooCommerce 3.0. (We were previously using a different enlargement plugin.)

I copied the title text and pasted it into the caption for each one of my product images in the Media Library. This caused the data-o_title clauses to be populated, which caused title text to appear in whatever Raven SEO Tools looks at when it crawls the product page, which caused my 125 image issues to go away. Problem solved ... Cindy
 

    On Tuesday, June 27, 2017 3:13 AM, Mike Jolley <notifications@github.com> wrote:
 

 I found a few resources to confirm this before replying, but it's not important. The ACTUAL images output to the page do have alt text and title. It's only the photoswipe image (which appears on click with javascript) which lacks these tags. This image is not important or parsed by search engines.http://community.theturninggate.net/viewtopic.php?id=7393
Such things are entirely unimportant to Photoswipe. The presentation is built on-the-fly from Javascript, and doesn't exist until the moment an image is clicked, so it's entirely irrelevant to SEO and sitemaps.
There was also a photoswipe issue confirming the same thing:dimsemenov/PhotoSwipe#757CC'ing @claudiosanches and @claudiulodro to commit this to memory in case it comes up again :)Thanks—

**Comment by mikejolley** [comment]
I think thats fixed in 3.1 FWIW :)

**Comment by shadtek** [comment]
@mikejolley 
I have PhotoSwipe - v4.1.3 - 2019-01-08 and my WooCommerce is version 3.7.0

The alt for images is important for WCAG 2.0
https://www.w3.org/TR/WCAG20-TECHS/F38
https://www.w3.org/TR/WCAG20-TECHS/F65
https://www.w3.org/TR/WCAG20-TECHS/H37
